ETABS_Engine/Modify/FlipEndPoints.cs

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2019, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using BH.oM.Structure.Elements;
+using BH.oM.Adapters.ETABS;
+using BH.oM.Adapters.ETABS.Elements;
+using BH.Engine.Structure;
+using BH.oM.Geometry;
+using BH.oM.Structure.Constraints;
+
+namespace BH.Engine.ETABS
+{
+    public static partial class Modify
+    {
+
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static Bar FlipEndPoints(this Bar bar)
+        {
+            // Flip the endpoints
+            Bar clone = bar.Flip();
+
+            // Flip orientationAngle
+            if (!bar.IsVertical())
```

Comment:
Generally working really well in terms of getting maintained orientation out. Found one corner case for a vertical bar:

![image](https://user-images.githubusercontent.com/22005920/73174303-b2e2b580-4107-11ea-86ac-42fa21d4eca4.png)